ophyd_basler/basler_camera.py

```diff
@@ -61,12 +64,23 @@ def __init__(self, *args, root_dir="/tmp/basler", cam_num=0, pixel_format="Mono8
         self.payload_size.put(self.camera_object.PayloadSize())
 
         # these are hardcoded for now, but should make them more flexible in the future
-        self.grab_timeout = 5000
         trigger_mode = "Off"
 
         self.camera_object.TriggerMode.SetValue(trigger_mode)
         self.camera_object.PixelFormat.SetValue(self.pixel_format)
 
+        # Exposure time can't be less than self.camera_object.ExposureTime.Min.
+        # We use seconds for ophyd, and microseconds for pylon:
+        min_exposure_us = self.camera_object.ExposureTimeAbs.Min
+        if min_exposure_us > 1e6 * self.exposure_time.get():
+            self.exposure_time.put(1e-6 * min_exposure_us)
+            warnings.warn(
+                f"Desired exposure time ({1e6 * self.exposure_time.get()} us) is less than \
+            the minumum exposure time ({min_exposure_us} us). Proceeding with minumum exposure time."
```

Comment:
I suggest splitting it into 2 separate lines with f-strings, as follows:
```suggestion
                f"Desired exposure time ({1e6 * self.exposure_time.get()} us) is less than "
                f"the minimum exposure time ({min_exposure_us} us). Proceeding with minimum exposure time."
```

Reply:
This also needs to be inside grab_images(), or else exposure_time.put() after __init__() won't do anything.